backbone/src/main/java/org/researchstack/backbone/onboarding/OnboardingSectionAdapter.java

```diff
@@ -0,0 +1,61 @@
+package org.researchstack.backbone.onboarding;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+
+import org.researchstack.backbone.model.survey.SurveyItem;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by TheMDP on 1/2/17.
+ */
+
+public class OnboardingSectionAdapter implements JsonDeserializer<OnboardingSection> {
+
+    ResourceNameJsonProvider mResourceNameConverter;
+
+    public OnboardingSectionAdapter(ResourceNameJsonProvider converter) {
+        mResourceNameConverter = converter;
+    }
+
+    @Override
+    public OnboardingSection deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+
+        JsonElement typeJson = json.getAsJsonObject().get(OnboardingSection.ONBOARDING_TYPE_GSON);
+        OnboardingSectionType type = context.deserialize(typeJson, OnboardingSectionType.class);
+
+        JsonElement resourceName = json.getAsJsonObject().get(OnboardingSection.ONBOARDING_RESOURCE_NAME_GSON);
+        if (resourceName != null) {
+            // Android does not support spaces or uppercase letters for resource names
```

Comment:
That's something we should maybe try to push back to iOS too and encourage them to stick to naming schemes that work on both platforms w/o adjustment